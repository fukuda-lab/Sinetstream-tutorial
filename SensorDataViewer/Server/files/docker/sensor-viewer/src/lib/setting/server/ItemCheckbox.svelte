<script lang="ts">
  import { setContext } from "svelte";
  import { itemKey } from "./server";

  export let name = "targets";
  export let value: string;

  setContext(itemKey, { getParent: () => value });
</script>

<div class="row">
  <label class="checkbox">
    <input type="checkbox" {name} {value} />
    <span />
  </label>
  <div><slot name="label" /></div>
</div>

<slot />
