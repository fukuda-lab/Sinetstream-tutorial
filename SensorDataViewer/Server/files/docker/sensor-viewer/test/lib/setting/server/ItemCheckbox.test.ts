import { cleanup, render } from '@testing-library/svelte';
import {
  afterEach, describe, expect, it,
} from 'vitest';

import ItemCheckbox from '../../../../src/lib/setting/server/ItemCheckbox.svelte';
import SlotTest from './SlotTest.svelte';

describe('SubItemLabel.svelt', () => {
  afterEach(() => cleanup());

  describe('name', () => {
    const value = 'item';

    it('default name', () => {
      const { container, getByRole } = render(ItemCheckbox, {
        props: { value },
      });
      expect(container.outerHTML).toMatchSnapshot();
      const item = getByRole('checkbox');
      expect(item).toBeTruthy();
      expect(item).toHaveProperty('name', 'targets');
    });

    it('name property', () => {
      const name = 'name1';
      const { container, getByRole } = render(ItemCheckbox, {
        props: { name, value },
      });
      expect(container.outerHTML).toMatchSnapshot();
      const item = getByRole('checkbox');
      expect(item).toBeTruthy();
      expect(item).toHaveProperty('name', name);
    });
  });

  it('value', () => {
    const value = 'item';
    const { container, getByRole } = render(ItemCheckbox, {
      props: { value },
    });
    expect(container.outerHTML).toMatchSnapshot();
    const item = getByRole('checkbox');
    expect(item).toBeTruthy();
    expect(item).toHaveProperty('value', value);
  });

  it('slot', () => {
    const value = 'item';
    const { container, getByTestId } = render(SlotTest, {
      props: {
        Component: ItemCheckbox,
        parameters: { value },
      },
    });
    expect(container.outerHTML).toMatchSnapshot();
    expect(getByTestId('default-slot')).toBeTruthy();
    expect(getByTestId('label-slot')).toBeTruthy();
  });
});
