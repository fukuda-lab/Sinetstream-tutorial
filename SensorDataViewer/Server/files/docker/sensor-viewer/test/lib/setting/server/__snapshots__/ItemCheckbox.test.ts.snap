// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubItemLabel.svelt > name > default name 1`] = `"<body><div><div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"item\\"> <span></span></label> <div></div></div> <!--<ItemCheckbox>--></div></body>"`;

exports[`SubItemLabel.svelt > name > name property 1`] = `"<body><div><div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"name1\\" value=\\"item\\"> <span></span></label> <div></div></div> <!--<ItemCheckbox>--></div></body>"`;

exports[`SubItemLabel.svelt > slot 1`] = `"<body><div><div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"item\\"> <span></span></label> <div><span data-testid=\\"label-slot\\" slot=\\"label\\">label slot</span></div></div> <div data-testid=\\"default-slot\\">default slot</div><!--<ItemCheckbox>--><!--<SlotTest>--></div></body>"`;

exports[`SubItemLabel.svelt > value 1`] = `"<body><div><div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"item\\"> <span></span></label> <div></div></div> <!--<ItemCheckbox>--></div></body>"`;
